clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp

```diff
@@ -367,6 +367,25 @@ class RenamerClangTidyVisitor
     return true;
   }
 
+  bool VisitDesignatedInitExpr(DesignatedInitExpr *Expr) {
+    for (const auto &Designator : Expr->designators()) {
+      if (!Designator.isFieldDesignator())
+        continue;
+      auto *FieldDecl = Designator.getFieldDecl();
+      if (!FieldDecl)
+        continue;
+      auto *II = FieldDecl->getIdentifier();
+      if (!II)
+        continue;
+      SourceRange FixLocation{
+          Designator.getFieldLoc(),
+          Designator.getFieldLoc().getLocWithOffset(II->getLength())};
```

Comment:
won't this getLocWithOffset point pass the end ?, usually end should point to last token, this will point to last token +1 character. In theory: `SourceRange FixLocation{Designator.getFieldLoc(), Designator.getFieldLoc());` should work in same way.

Reply:
You're right. Good to know!